ADAL/src/cache/ADTokenCache.m

```diff
@@ -567,7 +603,7 @@ - (BOOL)addOrUpdateImpl:(ADTokenCacheItem *)item
 
 - (NSString *)description
 {
-    return [NSString stringWithFormat:@"ADTokenCache: %@", _cache];
+    return [NSString stringWithFormat:@"ADTokenCache: %@", [self cacheCopy]];
```

Comment:
Creating a copy just to display it in a description is a tad wasteful, and creating that copy required you to use @synchronize anyways. And creating that copy will take longer then it took create the string...